### PR TITLE
Added more data to the table view.

### PR DESCRIPTION
**Pull request by wiseman**
Added call sign, speed, vertical speed, and number of messages to the
table view.

Instead of being a map from `ICAO` -> `AirplaneCoor`, `Airplanes` is now a
map from `ICAO` -> state vector, where the state vector (actually an
`AirplaneState` struct) holds coordinates, squawk code, call sign,
speed, vertical speed, etc.

The rows are starting to get a bit long; Eventually maybe there could
be a detail view with more data (autopilot, position accuracy, etc.).

### CODE REVIEW

**Comment by wiseman** [comment]
Screen Shot 2021-11-01 at 10 53 24 AM

File: apps/src/radar.rs [hunk 1]
```diff
@@ -315,19 +356,32 @@ fn main() {
...
                             .header(
-                                Row::new(vec!["ICAO Address", "Longitude", "Latitude", "Altitude"])
-                                    .bottom_margin(1),
+                                Row::new(vec![
+                                    "ICAO",
+                                    "Call sign",
+                                    "Longitude",
+                                    "Latitude",
+                                    "Altitude",
+                                    "   FPM",
```
Comment (wiseman) [inline apps/src/radar.rs:365]: Maybe there's a better way to do the label justification in TUI?

File: apps/src/radar.rs [hunk 2]
```diff
@@ -290,16 +311,36 @@ fn main() {
...
+                                lon = format!("{}", position.longitude);
+                                alt = format!("{}", altitude);
                             }
+                            rows.push(Row::new(vec![
+                                format!("{}", key),
+                                state.callsign.as_ref().unwrap_or(&empty).clone(),
+                                lat,
+                                lon,
+                                format!("{:>8}", alt),
+                                if let Some(vert_speed) = state.vert_speed {
```
Comment (wiseman) [inline apps/src/radar.rs:332]: I feel like there's a better way to do this that I'm missing.

Comment (wcampbell0x2a) [reply apps/src/radar.rs:332]: you could do:
```
state
    .vert_speed
    .map_or_else(|| "".to_string(), |v| format!("{:>6}", v)),
```

I'm not sure that's an improvement?

File: apps/src/radar.rs (hunk 1 above)
Comment (wcampbell0x2a) [reply apps/src/radar.rs:365]: I don't see any location to set this in the `tui::widgets::Cell`

File: apps/src/radar.rs [hunk 3]
```diff
@@ -159,9 +159,30 @@ fn main() {
             // decode
             if let Ok((_, frame)) = Frame::from_bytes((&bytes, 0)) {
                 if let DF::ADSB(ref adsb) = frame.df {
-                    if let ME::AirbornePositionBaroAltitude(_) = adsb.me {
-                        adsb_airplanes.add_extended_quitter_ap(adsb.icao, frame.clone());
-                    }
+                    adsb_airplanes.incr_messages(adsb.icao);
+                    match &adsb.me {
```
Comment (wcampbell0x2a) [inline apps/src/radar.rs:163]: I'd prefer:
```diff
-                        ME::NoPosition(_) => {},
                         ME::AircraftIdentification(identification) => {
                             adsb_airplanes.add_identification(adsb.icao, identification);
                         },
-                        ME::SurfacePosition(_) => {},
-                        ME::AirbornePositionBaroAltitude(altitude) => {
-                            adsb_airplanes.add_altitude(adsb.icao, altitude);
-                        },
                         ME::AirborneVelocity(vel) => {
                             adsb_airplanes.add_airborne_velocity(adsb.icao, vel);
                         },
-                        ME::AirbornePositionGNSSAltitude(altitude) => {
+                        ME::AirbornePositionGNSSAltitude(altitude)
+                        | ME::AirbornePositionBaroAltitude(altitude) => {
                             adsb_airplanes.add_altitude(adsb.icao, altitude);
                         },
-                        ME::Reserved0(_) => {},
-                        ME::SurfaceSystemStatus(_) => {},
-                        ME::Reserved1(_) => {},
-                        ME::AircraftStatus(_) => {},
-                        ME::TargetStateAndStatusInformation(_) => {},
-                        ME::AircraftOperationalCoordination(_) => {},
-                        ME::AircraftOperationStatus(_) => {},
+                        _ => ()
```

File: apps/src/lib.rs [hunk 4]
```diff
@@ -42,35 +69,56 @@ impl Airplanes {
...
-                }
-            }
+    pub fn add_altitude(&mut self, icao: ICAO, altitude: &Altitude) {
+        let state = self.0.entry(icao).or_insert_with(AirplaneState::default);
+        let now = SystemTime::now();
+        match altitude.odd_flag {
+            CPRFormat::Odd => {
+                state.coords = AirplaneCoor {
+                    altitudes: [state.coords.altitudes[0], Some(*altitude)],
+                    last_time: SystemTime::now(),
```
Comment (wcampbell0x2a) [inline apps/src/lib.rs:104]: use `now` variable set earlier

**Comment by wcampbell0x2a** [comment]
Awesome, what an improvement.

**Pull request merged**